modules/chatgpt.py

```diff
@@ -0,0 +1,149 @@
+from api.gooseai import GooseAI
+from api.openai import OpenAI
+from config import (
+    CONFUSED_RESPONSE,
+    openai_api_key,
+)
+from modules.module import Module, Response
+from utilities.serviceutils import ServiceMessage
+from servicemodules.serviceConstants import service_italics_marks, default_italics_mark
+from servicemodules.discordConstants import rob_id, stampy_id
+import openai
+import re
+
+openai.api_key = openai_api_key
+
+
+class ChatGPTModule(Module):
+    def __init__(self):
+        super().__init__()
+        self.class_name = "ChatGPTModule"
+
+        self.message_logs = {}  # one message log per channel
+        self.log_max_messages = 15  # don't store more than X messages back
+        self.log_max_chars = 1500  # total log length shouldn't be longer than this
+        self.log_message_max_chars = (
+            1000  # limit message length to X chars (remove the middle part)
+        )
+        self.openai = OpenAI() if openai_api_key else None
+        if not openai_api_key:
+            self.log.info(
+                self.class_name,
+                warning="No OpenAI API key found in env",
+            )
+
+    def process_message(self, message: ServiceMessage) -> Response:
+        self.message_log_append(message)
+
+        if message.is_dm:
+            if message.author.id != rob_id:
+                self.log.info(
+                    self.class_name,
+                    author=message.author.id,
+                    author_type=type(message.author.id),
+                )
+                return Response()
+
+        if not self.is_at_me(message):
+            return Response()
+
+        return Response(
+            confidence=3, callback=self.chatgpt_chat, args=[message], kwargs={}
```

Comment:
not very confident in chatgpt? in any case, do we have some guideline what number to use for new modules?

Reply:
The problem is ChatGPT will *always* have a response of some kind, so anything lower confidence than it can never respond. Hence, give other modules a chance to go first.
Confidence tuning is kind of an artistic choice so it's hard to write guidelines for, but I can try

Reply:
`confidence=randrange(1, 6)`  :trollface: